### PR TITLE
feat(Menu): Drilldowns

### PR DESCRIPTION
**Pull request by jschuler**
**What**: Closes https://github.com/patternfly/patternfly-react/issues/5024

https://patternfly-react-pr-5321.surge.sh/components/menu#with-drilldown

### CODE REVIEW

File: packages/react-core/src/components/Menu/Menu.tsx [hunk 1]
```diff
@@ -26,6 +26,10 @@ export interface MenuProps extends Omit<React.HTMLAttributes<HTMLDivElement>, 'r
   'aria-label'?: string;
   /** Indicates if menu contains a flyout menu */
   containsFlyout?: boolean;
+  /** Indicates if menu contains a drilldown menu */
+  containsDrilldown?: boolean;
+  /** Indicates if a menu is drilled into */
+  drilledIn?: boolean;
```
Comment (tlabaj) [inline packages/react-core/src/components/Menu/Menu.tsx:32]: perhaps rename to `isMenuDrilledin`

File: packages/react-core/src/components/Menu/MenuItem.tsx [hunk 2]
```diff
@@ -40,6 +42,12 @@ export interface MenuItemProps extends Omit<React.HTMLProps<HTMLLIElement>, 'onC
   flyoutMenu?: React.ReactNode;
   /** Callback function when mouse leaves trigger */
   onShowFlyout?: (event?: any) => void;
+  /** Drilldown menu */
+  drilldownMenu?: MenuProps;
+  /** Sub menu direction */
+  direction?: 'down' | 'up';
+  /** True if item is on current selection path */
+  onPath?: boolean;
```
Comment (tlabaj) [inline packages/react-core/src/components/Menu/MenuItem.tsx:50]: `isOnPath`?

**Review by tlabaj — COMMENTED** [review]
can you add integration test for the new props.

**Review by mcarrano — COMMENTED** [review]
This is looking good @jschuler . Just two comments/questions:

* I love the animation when you open a new level. I see this when I go from level 1 to 2, but not when I drill down to the third level. Was it intended that these behave differently?

* I'm wondering what the alignment of the description text should be in the breadcrumb, or if it should display there at all. I'm referring to this-

Screen Shot 2021-01-20 at 2 32 59 PM

@doruskova Please also take a look at this. Interested in your thoughts about the description text question above.

**Comment by doruskova** [comment]
@mcarrano I think it should be aligned under the name of the category. I can imagine that the description can be helpful if your data can be described in 1 sentence. 
Snímek obrazovky 2021-01-21 v 17 27 19

**Comment by mcarrano** [comment]
I agree with that @doruskova .

@jschuler is it possible to make that change?

**Comment by kmcfaul** [comment]
@mcoker I think the update to get the description in-line with the group title might be a core change? Thoughts?

I've got the PR pulled and will make any updates

**Comment by mcarrano** [comment]
@kmcfaul @mcoker I'm not even sure it's necessary to have the description line in the title like this. If it's not in Core, then I'm guessing it wasn't part of the initial design. It must be possible for menu items to have descriptions, however. @doruskova what are your thoughts? Do you think this adds value or clutter?

**Comment by mcoker** [comment]
@mcarrano @kmcfaul it would require a core change if we decide to keep the description in the drill up item.

**Comment by mcarrano** [comment]
@kmcfaul @mcoker In that case I'd just rather not carry the description into the "drill-up" label. Can we do that?

**Comment by mcoker** [comment]
@kmcfaul can we just not show the description in a drill up item in the react component?

**Comment by kmcfaul** [comment]
Yes I updated to remove the menu header's descriptions in my PR addition (which is referenced above on my fork).

Also, do you mind taking a look at the drilldown's transition animation @mcoker? It only appears to be working for the first drilldown but not for subsequent inner menus. I'm not sure if I'm missing a styling, or if we need to re-apply a class somewhere (because once `pf-m-drilled-in` is applied it remains there for the following drilled in menus without change).

**Comment by mcarrano** [comment]
Thanks for removing the description from the header @kmcfaul . I had the same question about the animation when I looked at this first a month ago. I think either it should be made to work for all levels or we should remove it.

**Comment by tlabaj** [comment]
closing this PR.  The issue will be resolved in PR #5485

**Pull request closed without merging**